packages/flutter/lib/src/material/search.dart

```diff
@@ -54,12 +60,14 @@ Future<T?> showSearch<T>({
   required BuildContext context,
   required SearchDelegate<T> delegate,
   String? query = '',
+  bool useRootNavigator = false,
```

Comment:
all the other showdialog methods default useRootNavigator to true. In the most case, developer would want to see the search cover the entire screen i assume?

Reply:
I agree. however, it will pose a small problem for people who are intentionally wanting it to show up on a navigator other than rootNavigator, although this is very rare.

Reply:
if we set the default value to true, it will be possible to change the behavior in their app.

Reply:
I think this is relative small risk because like you said, it is rare. If people do care about this, they should add their test in flutter/test. I suggest we set it to true and see if there is any breakage in customer test. It also unifies the API between show dialogs and showsearch

Reply:
Indeed. I just change it to true by default

Reply:
it seems all the tests at `test/material/search_test.dart` passed